home/private_dot_local/lib/agent-equipment/agent_equipment/discovery.py

```
@@ -0,0 +1,1509 @@
+def _request_is_valid(value: object) -> bool:
+    if type(value) is not EquipmentDiscoveryRequest:
+        return False
+    if (
+        type(value.command) is not str
+        or type(value.harness) is not str
+        or type(value.request_digest) is not str
+        or value.targets is not None
+        and (
+            type(value.targets) is not tuple
+            or any(type(target) is not str for target in value.targets)
+        )
```

Ruff (0.16.1)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)